### DISCUSSION TITLE
API Rules: filter based on time

### DISCUSSION THREAD

**Discussion by nq** [pocketbase/pocketbase#3438] [Q&A]
I'm trying to give access to a collection for users who got a timestamp in the future.
Ex.
```
{
      "id": "RECORD_ID",
      "collectionId": "COLLECTION_ID",
      "collectionName": "users",
      "validUntil": "2023-11-01 23:59:59.456Z"
      .... (more data)
    },
```
    
The API Filter rules on the target collection are following
List/Search & View:
`@request.auth.validUntil > @now`

as a result i only get: ClientResponseError 404: The requested resource wasn't found.

**Comment by ganigeorgiev** [comment]
Are you sure you are authenticated?
If you are sending the request with the browser could you post the request headers of the list or view API request from the _"Inspect tool > Network"_ tab?

**Reply by ganigeorgiev** [reply]
Note that if you want to target all users (not just the authenticated one) you can use directly the field in the condition (_assuming that the above rule is set as list/view API rule of the users collection_), aka:

```js
validUntil > @now
```

**Reply by nq** [reply]
thanks for the suggestion and quick answer! issue is due to the user not being authenticated. trying currently to solve the issue. Im using NextJS 13 (app router) on server side components.
